PED/SetIkTarget.md

```diff
@@ -5,18 +5,68 @@ ns: PED
 
 ```c
 // 0xC32779C16FCEECD9 0x6FE5218C
-void SET_IK_TARGET(Ped ped, int ikIndex, Entity entityLookAt, int boneLookAt, float offsetX, float offsetY, float offsetZ, Any p7, int blendInDuration, int blendOutDuration);
+void SET_IK_TARGET(Ped ped, int ikIndex, Entity entityLookAt, int boneLookAt, float offsetX, float offsetY, float offsetZ, int ikTargetFlags, int blendInDuration, int blendOutDuration);
 ```
 
+Sets the IK target for a given IK part belonging to the ped. The IK target will only be valid for one update, so it needs to be set for as long as it is needed (to avoid IK targets not being cleared and getting stuck enabled).
+
 ## Parameters
-* **ped**: 
-* **ikIndex**: 
-* **entityLookAt**: 
-* **boneLookAt**: 
-* **offsetX**: 
-* **offsetY**: 
-* **offsetZ**: 
-* **p7**: 
+* **ped**: The ped handle.
+* **ikIndex**: See ``IK_PART``.
+* **entityLookAt**: Set to 0 for no entity, offsets will be world coordinates instead.
+* **boneLookAt**: Set to -1 for no target bone.
+* **offsetX**: X-axis offset calculated from the position of ``entityLookAt``.
+* **offsetY**: Y-axis offset calculated from the position of ``entityLookAt``.
+* **offsetZ**: Z-axis offset calculated from the position of ``entityLookAt``.
+* **ikTargetFlags**: See ``IK_TARGET_FLAGS``.
 * **blendInDuration**: 
```

Comment:
No description for this one? :(

Reply:
I could not find out the thing that it does, in my testing it seemed to have no effect on anything

Reply:
Are they even named right these look like they would be similar to other animation flags (aka [TaskPlayAnim](https://docs.fivem.net/natives/?_0xEA47FE3719165B94))

These are probably meant to be `blendInSpeed` and `blendOutSpeed`

Reply:
Well the old param is still good, it blend in time.

And `blendOutDuration` will set the time to blend out.

I'll do some tests to confirm.

Reply:
So after some tests and research, this is what I found:

`-1` seems to be the default values, based on [decompiled files](https://github.com/root-cause/v-decompiled-scripts/blob/afd9d9e3235fe6411c9d2c5ad7109a9ab279d042/re_duel.c#L5329).

There's other cases where R* use to set a specific time, [like here](https://github.com/root-cause/v-decompiled-scripts/blob/afd9d9e3235fe6411c9d2c5ad7109a9ab279d042/country_race_controller.c#L811).

When testing, I see that it works perfectly when setting specific time to blend out and `-1` to blend in
![image](https://github.com/user-attachments/assets/fabda0b7-a177-45cd-82bc-a5263a7d17d3)

But you can see here, when testing `SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, -1, -1)`

https://github.com/user-attachments/assets/04bc6efa-ba57-4ee9-9f15-6257f2fec570

So I guess it's only to be used with a specific blend out in `ms`.

Reply:
so what should I write for it?